sigh/lib/sigh/runner.rb

```diff
@@ -88,19 +88,34 @@ def fetch_profiles
         includes += ',certificates'
       end
 
+      # also include devices if specific device udids were specified
+      force_profile_device_udids = []
+      unless Sigh.config[:force_profile_device_udids].nil?
+        includes += ",devices"
+        force_profile_device_udids = Sigh.config[:force_profile_device_udids].split(',').map(&:strip)
+      end
+
       results = Sigh.config[:cached_profiles]
       results ||= Spaceship::ConnectAPI::Profile.all(filter: filter, includes: includes)
       results.select! do |profile|
         profile.bundle_id.identifier == Sigh.config[:app_identifier]
       end
 
       results = results.find_all do |current_profile|
+        profile_qualifies = false
         if current_profile.valid? || Sigh.config[:force]
-          true
+          profile_qualifies = true
+          if force_profile_device_udids.length > 0
+            filtered_devices = current_profile.devices.select { |device| force_profile_device_udids.include?(device.udid) }
+            if force_profile_device_udids.length != filtered_devices.length
+              UI.message("Provisioning Profile '#{current_profile.name}' is valid but does not match force_profile_device_udids skipping this one...")
```

Comment:
UI.message("Provisioning Profile '#{current_profile.name}' is valid but does not match force_profile_device_udids, skipping this one...")

Reply:
I suppose that it's about missing comma: `force_profile_device_udids, skipping this one..."

Reply:
ah... let me do that.

Reply:
addressed in 7f8120179e08d57f91c3655f18fcf8806a756f5f